http/openapi.go

```diff
@@ -0,0 +1,509 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package http
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/getkin/kin-openapi/openapi3gen"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+// OpenApiSpec contains the OpenAPI specification for DefraDB.
+var OpenApiSpec = &openapi3.T{
+	OpenAPI: "3.0.3",
+	Info: &openapi3.Info{
+		Title:   "DefraDB API",
+		Version: "0",
+	},
+	Paths: make(openapi3.Paths),
+	Servers: openapi3.Servers{
+		&openapi3.Server{
+			Description: "Local DefraDB instance",
+			URL:         "http://localhost:9181/api/v0",
+		},
+	},
+	ExternalDocs: &openapi3.ExternalDocs{
+		Description: "Read more about DefraDB",
+		URL:         "https://docs.source.network",
+	},
+}
+
+func init() {
```

Comment:
suggestion: We will now have 3 different files for doing this, `cli.go`, `handler.go`, and `openapi.go`. Working with anyone of them in isolation is error prone, working with all of them distributed as they currently are is even more so.  It also feels quite important to me that they all follow the same structure (as much so as is reasonable).

Could you perhaps spend some time now looking at pulling these 3 files out to their own shared package? Called `api` or something. And provide us with a single location in which we have to maintain these?

Longer term (I don't want to pressure you now) we could perhaps unify them further so that instead of splitting the functions in that new package by client, we could instead split them by function, for example (very roughy):
```
func BindTxCreate() {
  MakeTxCreateCommand(cfg)
  
  tx.Post("/", tx_handler.NewTxn)
  
  txnCreate := openapi3.NewOperation()
  txnCreate.AddParameter(txnReadOnlyQueryParam)
  txnCreate.AddResponse(200, txnCreateResponse)
  txnCreate.AddResponse(400, errorResponse)
}

func BindPatchSchema() {
... etc
```

Reply:
I think I've found a solution that will make the http routes simpler. I've moved the route registration to bind functions which now require OpenAPI definitions for each route. This should help enforce that all endpoints have basic descriptions. Validating client requests in the tests will be the final step to enforce the correctness of the definitions.